app/controllers/concerns/authentication.rb

```diff
@@ -21,6 +21,10 @@ def user_signed_in?
 
   def login(user)
     Current.user = user
+
+    # Sync accounts to ensure that exchange rates are up-to-date
+    Current.family.sync_accounts
```

Comment:
Unfortunately, it won't work at the `application_controller` level because `Current.user` wil be always `nil` as is set up after in the child controllers with `before_action :authenticate_user!`:

https://github.com/maybe-finance/maybe/blob/b3c48d13e70780caa146cdb79fce6bb851fabd76/app/controllers/accounts_controller.rb#L1-L5

We should sync accounts after the user is found and before `last_login_at` is set with a new value, somenthing like: 

```ruby
module Authentication
  # ...

  def authenticate_user!
    if user = User.find_by(id: session[:user_id])
      Current.user = user

      if user.last_login_at.nil? || user.last_login_at < Date.current.beginning_of_day
        Current.family.sync_accounts
      end

      user.update(last_login_at: DateTime.now)
    else
      redirect_to new_session_url
    end
  end

  # ...
end
``` 

`authenticate_user!` is not the best place to put this logic, maybe it's more clear to move it at the controller level:

```ruby
class PagesController < ApplicationController
  before_action :authenticate_user!
  before_action :sync_accounts
  after_action :set_user_last_login
end
```

Another approach is to move everything into a cron job that runs once a day.

Reply:
What about lifting up authentication to happen at the application controller level? (and ignoring session routes)

That would probably require a separate PR, but I'm just trying to think through the best long-term option here.

I agree that we should avoid doing this in the authentication module.

Reply:
Authentication is now centralized per #598 

Let me know if you want to tackle this.  If not, happy to create a separate issue for it!

Reply:
I saw your PR, seems all good!
I'd like to tackle this, just a recap:

- We need to add `last_login_at` to the users table;
- At the `application_controller` level we sync user's accounts with the method `sync_accounts` above.

Let me know if I'm missing anything!

Reply:
Great, will keep this PR open—feel free to request another review when you're at a good stopping point.

And yes, that logic makes sense to me overall. The biggest focus I think is to get a "sketch" of the overall flow in place that will be easy to iterate on as we add more and more to the sync logic and process.  For example, #596 will cover some of those comments you had mentioned about global sync alerts, and we'll also be adding in error handling for the sync process in the future too.

Reply:
You can find the updated code, I think that's all good now!